scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -455,7 +455,41 @@ static s32 pick_idle_cpu(struct task_struct *p, struct task_ctx *taskc,
 			goto found_cpu;
 		}
 
-		// TODO: handle affinitized tasks to be more topology aware
+		// First try to find an idle SMT in the LLC
+		if (smt_enabled && taskc->mask && llcx->cpumask) {
+			bpf_cpumask_and(taskc->mask, cast_mask(llcx->cpumask),
+					p->cpus_ptr);
+			if (taskc->mask &&
```

Comment:
Maybe you can move taskc->mask in a variable and check it once if it's NULL, the verifier should be happy.